yarn-project/sequencer-client/src/slasher/slasher_client.ts

```diff
@@ -105,21 +86,23 @@ export class SlasherClient extends WithTracer {
 
   constructor(
     private config: SlasherConfig & L1ContractsConfig & L1ReaderConfig,
-    private store: AztecAsyncKVStore,
     private l2BlockSource: L2BlockSource,
+    private epochCache: EpochCache,
     telemetry: TelemetryClient = getTelemetryClient(),
     private log = createLogger('slasher'),
   ) {
     super(telemetry, 'slasher');
 
-    this.blockStream = new L2BlockStream(l2BlockSource, this, this, createLogger('slasher:block_stream'), {
-      batchSize: config.blockRequestBatchSize,
-      pollIntervalMS: config.blockCheckIntervalMS,
-    });
-
-    this.synchedBlockHashes = store.openMap('slasher_block_hashes');
-    this.synchedLatestBlockNumber = store.openSingleton('slasher_last_l2_block');
-    this.synchedProvenBlockNumber = store.openSingleton('slasher_last_proven_l2_block');
+    this.blockStream = new L2BlockStream(
+      this.l2BlockSource,
+      this.l2BlockSource,
```

Comment:
I don't think this is going to do what you want. The blockstream works by comparing two sources of data, a "remote" (archiver) and a "local" (current service) and pulling data from one into the other. If you set the same data source for both, it will never trigger anything, since there won't be any difference between the two.

If you want to simplify the existing code, you can use the `L2TipsStore` here, that given a `store` it keeps a local copy of whatever the block stream passed in.